sihl_core/lib/contract.ml

```diff
@@ -1,3 +1,34 @@
-module type EMAILTRANSPORT = sig
-  val send : Email_core.t -> (unit, string) result Lwt.t
+module Email = struct
+  module type TRANSPORT = sig
+    val send : Email_core.t -> (unit, string) result Lwt.t
+  end
+end
+
+module Migration = struct
+  module type MIGRATION = sig
+    val migration : unit -> Db_migration_core.migration
+  end
+
+  module type REPOSITORY = sig
+    val create_table_if_not_exists :
+      (module Caqti_lwt.CONNECTION) ->
+      unit ->
+      (unit, [> Caqti_error.call_or_retrieve ]) Result.t Lwt.t
+
+    val get :
+      (module Caqti_lwt.CONNECTION) ->
+      namespace:string ->
+      ( Db_migration_core.Model.t option,
+        [> Caqti_error.call_or_retrieve ] )
+      Result.t
+      Lwt.t
+
+    val upsert :
```

Comment:
Btw, upsert is something I don't like at all as it does not fit into change set thinking at all and I think we should design these APIs from change set point of view where you really need to know if you create new record or change an existing one.

Reply:
We don't have to require users of these APIs to use changes sets from day one, but we should not design APIs that prevent us from using that approach.

Reply:
I think the change set thinking would make a lot of sense on Service/HTTP API layer, but I think we can go there once we have some ppx knowledge as this requires us to write a ton of boilerplate because of the static types. 

I don't see how introducing change sets on Repo Layer makes our life easier now. Later on we can easily change that without breaking the Service API.